android-json-form-wizard/src/main/java/com/vijay/jsonwizard/utils/Utils.java

```diff
@@ -821,7 +822,20 @@ public static JSONObject getRepeatingGroupCountObj(@NotNull WidgetArgs widgetArg
         return repeatingGroupCountObj;
     }
 
-
+    public static JSONObject generateTranslatableValue(String value, JSONObject item) throws JSONException {
+        FormUtils formUtils = new FormUtils();
+        JSONObject newValue = new JSONObject();
+        if (item.has(JsonFormConstants.OPTIONS_FIELD_NAME)) {
+            JSONArray options = item.getJSONArray(JsonFormConstants.OPTIONS_FIELD_NAME);
+            JSONObject selectedOption = formUtils.getOptionFromOptionsUsingKey(options, value);
+            newValue.put(JsonFormConstants.VALUE, value);
+            newValue.put(JsonFormConstants.TEXT, selectedOption.optString(JsonFormConstants.TRANSLATION_TEXT, ""));
+            return newValue;
+        }
+        newValue.put(JsonFormConstants.VALUE, value);
+        newValue.put(JsonFormConstants.TEXT, item.optString(JsonFormConstants.TRANSLATION_TEXT, ""));
+        return newValue;
+    }
```

Comment:
Please optimize this method. Line `825` to `838` and `985` to `997` can be in the same method

Reply:
Done